MetaMask/metamask-mobile
# fix: aggregated balance calculation

## **Description**

Fixes the calculation for how much the aggregated balance changed in the last 1 day.

The fiat price 1 day ago was calculated incorrectly.  For example it would calculate:

```json
{"ethFiat1dAgo": 1.2011985977032296, "ethFiat": 1.28, "pricePercentChange1d": -6.156359554435198}
```

But this is backwards.  If the price went down 6% in the last 24 hours, you would have had *more* fiat $ in the past, not less.   This broke further calculations. 

## **Related issues**
#10406 

## **Manual testing steps**

1. Open a wallet with just eth, no erc20 tokens (or hide them all)
2. Verify the aggregated fiat balance % matches the balance % for ETH in the token list

## **Screenshots/Recordings**



### **Before**

Simplest case: a wallet with just eth.  The aggregated balance should match ETH since it's the only token:

<img width="345" alt="Screenshot 2024-07-24 at 6 46 20 PM" src="https://github.com/user-attachments/assets/11f22fc7-abe8-46db-87d4-37f86fc8986d">



### **After**

<img width="344" alt="Screenshot 2024-07-24 at 6 45 54 PM" src="https://github.com/user-attachments/assets/7c982c95-ebce-437b-a394-f5b6d0d4601d">

Matches extension now:

<img width="878" alt="Screenshot 2024-07-24 at 6 41 16 PM" src="https://github.com/user-attachments/assets/c9c67a3e-07a9-4ea0-ad1a-d87dc37ca387">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
